igor-web/src/main/java/com/netflix/spinnaker/igor/scm/AbstractCommitController.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.scm;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import groovy.transform.InheritConstructors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+
+@RestController
+public abstract class AbstractCommitController {
+    private static final Logger log = LoggerFactory.getLogger(AbstractCommitController.class);
+    protected ExecutorService executor;
+    protected ObjectMapper objectMapper;
+
+
+    @ResponseStatus(value = HttpStatus.BAD_REQUEST, reason = "toCommit and fromCommit parameters are required in the query string")
+    @InheritConstructors
+    public static class MissingParametersException extends RuntimeException {
```

Comment:
This _can_ be private, it's not used anywhere else. But it _might_ make sense to keep it public for future use though.